### PR TITLE
#169817865 SignIn (Email and Password)

### PR DESCRIPTION
**Pull request by Jajabenit250**
#### What does this PR do?
- User Should SignIn with Email and Password
#### Description of Task to be completed?
- User Input his/ her Email and Password 
- If email and Password match to what is registered to our database user will get success message
- If the Input doesn't match to registered one user will get error message ("Email and password not match")
- User has to input valid Email and Password. e.g: (example@jaja.com, 123456790889).
#### How should this be manually tested?
-  Clone this repository
-  git checkout feature/169817865/User-SignIn
-  open console and type and run(- npm install) to install all dependencies
-  and then run (- npm start) to start the server
#### Any background context you want to provide?
- User need to be verify his email before sign In 
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/169817865
#### Screenshots (if appropriate) 
Screen Shot 2019-12-09 at 19 10 58

#### Questions:

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 import swagger from './routes/swagger';
+import userRoute from './routes/userRoute';
 
 const app = express();
 app.use(express.json());
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 
+
 app.use('/API/v1', swagger);
```
Comment (aanchirinah) [inline src/app.js]: ### Nitpick

- Maintain consistency with string casing, preferably should be of the lower case.

File: src/controllers/userController.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+   * Logs in a user by checking if they exist in the database
+   * and if the supplied password matches the stored password
+   * @param {Object} req The request object
+   * @param {Object} res The response object
+   * @returns {Object} A user object with selected fields
+   * excluing the password
+   */
+  static async signIn(req, res) {
+    try {
+      checkEmailpassword(req, res);
```
Comment (aanchirinah) [inline src/controllers/userController.js]: ### Nitpick

- I see `Async` but no `Await` block of code being used.

File: src/controllers/userController.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    } catch (e) {
+      return response.errorMessage(
+        req,
```
Comment (aanchirinah) [inline src/controllers/userController.js]: ### Nitpick

- Handle error messages in this format
```
- Error code
- Message
- Data
- StatusCode
```

File: src/database/models/user.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+
+module.exports = (sequelize, DataTypes) => {
+  const user = sequelize.define('user', {
+    firstName: DataTypes.STRING,
```
Comment (aanchirinah) [inline src/database/models/user.js]: ### Nitpick

- Add `allowNull` property

**Review by aanchirinah — CHANGES_REQUESTED** [review]
Great work please attend to the changes requested

**Review by jo-rdan — APPROVED** [review]
Good job Benit, keep it up!

**Comment by Jajabenit250** [comment]
@aanchirinah All change requested are resolved

**Review by aanchirinah — APPROVED** [review]

**Review by shemaeric1234 — APPROVED** [review]

**Pull request merged**